src/System Application/App/Password/src/PasswordDialog.Page.al

```diff
@@ -85,12 +95,16 @@ page 9810 "Password Dialog"
     var
         PasswordDialogImpl: Codeunit "Password Dialog Impl.";
         PasswordMismatchErr: Label 'The passwords that you entered do not match.';
+        PasswordSameAsOldErr: Label 'The new password cannot be the same as the old password.';
+        OldPasswordMismatchErr: Label 'The old password does not match the entered password.';
         [NonDebuggable]
         PasswordValue: Text;
         [NonDebuggable]
         ConfirmPasswordValue: Text;
         [NonDebuggable]
         OldPasswordValue: Text;
+        [NonDebuggable]
+        OldPasswordToCompare: Text;
```

Comment:
why are these not SecretText?

Reply:
@grobyns
Actually the reason is that on the page the user cannot interact directly with SecretText, variables with type text are used for input. Therefore, it was more convenient to compare text with text at once. 

For the same reason, I decided to place the logic of checks directly on the page.
Anyway, I'll fix what you asked for )